src/server/api/assignment.js

```diff
@@ -48,7 +48,15 @@ function getContacts(assignment, contactsFilter, organization, campaign) {
     }
 
     if (contactsFilter.hasOwnProperty('messageStatus') && contactsFilter.messageStatus !== null) {
-      query = query.where('message_status', contactsFilter.messageStatus)
+      if (pastDue && contactsFilter.messageStatus === 'needsMessage') {
+        query = query.where('message_status', '')
```

Comment:
this should be `query = query.whereIn('message_status', ['', 'needsMessage'])` at least.

When does messageStatus start or end up as `""`?  That seems like a bug to me (wherever it is), and we should make sure message_status always has a value, so we don't need to test for `""`.